overlord/snapstate/autorefresh.go

```diff
@@ -974,3 +974,127 @@ func MockRefreshCandidate(snapSetup *SnapSetup) interface{} {
 		SnapSetup: *snapSetup,
 	}
 }
+
+func incrementSnapRefreshFailures(st *state.State, snapsup *SnapSetup) error {
+	var snapst SnapState
+	err := Get(st, snapsup.InstanceName(), &snapst)
+	if err != nil {
+		return err
+	}
+
+	// Update refresh failure information for snap revision
+	if snapst.RefreshFailures == nil {
+		// Initialize RefreshFailures
```

Comment:
This comment doesn't help much. 

I would say something like this: in the case of the initial failure the zero value of the refresh failure structure is used in the else case below.

Reply:
I think instead of this comment and the one preceeding the `if` I would replace them with one single comment explaining in what case this will be nil